theano/compile/nanguardmode.py

```diff
@@ -106,6 +106,8 @@ def contains_nan(arr, node=None, var=None):
     """
     if not _is_numeric_value(arr, var):
         return False
+    elif "int" in str(getattr(arr, 'dtype', '')):
```

Comment:
You're still missing Booleans. Why don't you want to use `getattr(arr, 'dtype', '') in T.discrete_dtypes`?